decoder/src/log.rs

```diff
@@ -27,31 +27,38 @@ pub fn log_defmt(
     line: Option<u32>,
     module_path: Option<&str>,
 ) {
-    let level = match frame.level() {
-        crate::Level::Trace => Level::Trace,
-        crate::Level::Debug => Level::Debug,
-        crate::Level::Info => Level::Info,
-        crate::Level::Warn => Level::Warn,
-        crate::Level::Error => Level::Error,
-    };
-
     let timestamp = frame
         .display_timestamp()
         .map(|display| display.to_string())
         .unwrap_or_default();
-    let target = format!("{}{}", DEFMT_TARGET_MARKER, timestamp);
     let display = frame.display_message();
 
-    log::logger().log(
-        &Record::builder()
-            .args(format_args!("{}", display))
-            .level(level)
-            .target(&target)
-            .module_path(module_path)
-            .file(file)
-            .line(line)
-            .build(),
-    );
+    if frame.level().is_some() {
+        let level = match frame.level().expect("log level expected") {
```

Comment:
nit: you could avoid the `expect` here by doing an `if let`, instead of `if`, in the preceding line.

Reply:
Good catch.